compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -465,6 +466,49 @@ static LogicalResult setRootConfigForConvDecomposePipeline(
     tileSizeLevel0 = {0, OC, 4, OW, 0, 0, 0};
     tileSizeLevel1 = {1, OC, 1, OW, 0, 0, 0};
     tileSizeLevel2 = {0, 0, 0, 0, IC, 1, 1};
+  } else if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(linalgOp)) {
+    // Notes:
+    // =====
+    // An inherent property of depthwise convolutions is that they cannot be
+    // expressed in terms of matmuls, unlike the above (dense) conv-2ds. The
+    // tile sizes we choose below are therefore not constrained by the AIE
+    // matmul instructions.
+    //
+    // The logic is very fragile, and there are no guard-rails: there are no
+    // checks that the data tiles are not too large, or that the input
+    // dimensions are perfectly tiled by the hard-coded tile dimensions below.
+    // These will be done as a follow-up task.
+
+    // Outer most scf.forall tiling. Defines the sizes of data tiles in L2
+    // (shared memory). Specifically, with the values selected below, the output
+    // data N x 4 x 16 elements. The input data has N x (4 + kh) x (1 + kw) x 16
+    // elements. The kernel has kh x kw x 16 elements.TODO(newling)
+    // 1) check that the output tile perfectly tiles the input image
+    // 2) check that the sum of the L2 (memtile)  allocations is within memory
+    // budget
+    tileSizeLevel0 = {
+        /* N */ 0,        /* output width */ 4,  /* output height */ 1,
+        /* channel */ 16, /* kernel height */ 0, /* kernel width */ 0};
+
+    // Inner-most scf.forall tiling. The iteration space corresponds to
+    // individual AIE cores. TODO(newling)
+    // 1) check that the core array of shape Nx(16/4) is valid. N columns?
+    tileSizeLevel1 = {
+        /* N */ 1,       /* output width */ 0,  /* output height */ 0,
+        /* channel */ 4, /* kernel height */ 0, /* kernel width */ 0};
```

Comment:
Similar to Conv2d channel last, I think we should first try tile sizes as

tileSizeLevel0 = {0, 4, OW, OC, 0, 0};
tileSizeLevel1 = {1, 1, OW, OC, 0, 0};
tileSizeLevel2 = {0, 0, 0, 0, 1, 1};

If the batch size = 1, we could use 4x4 array, which means OW (OC) in the first level can be 4*OW (4*OC).